### ISSUE TITLE
css-paint-api/paint2d-image.html reftest may snapshot too early

### ISSUE THREAD

**Issue by asajeffrey** [w3c/web-platform-tests#6492]
In the css-paint-api/paint2d-image.html reftest (https://github.com/w3c/web-platform-tests/blob/master/css-paint-api/paint2d-image.html) AFAICT there is nothing stopping the reftest from taking a snapshot before the html5.png image has loaded. The image is mentioned in a css-border-image property, but this doesn't delay the reftest snapshot. If two rAFs happen (https://github.com/w3c/web-platform-tests/blob/master/common/css-paint-tests.js) after the module has loaded but before the image has loaded, the reftest will fail.

**Comment by asajeffrey** [comment]
cc @bfgeek

**Comment by gsnedders** [comment]
The screenshot should always be taken after the load event has fired, and the load event waits for `border-image` subresources in all browsers except for Edge. Unless the Houdini API causes something weird to happen with that?

**Comment by asajeffrey** [comment]
All browsers except for Edge and Servo :/ Does the spec require load events to be delayed until CSS images are loaded?

**Comment by gsnedders** [comment]
No, it doesn't. (The HTML spec defers to CSS, and CSS makes no statement, which is https://github.com/w3c/csswg-drafts/issues/1088.)

**Comment by gsnedders** [comment]
(This is also unfortunate because there's no way to actually wait on things like `border-image` loading otherwise, given in the different-origin case you can't even detect it using the resource loading API.)

**Comment by asajeffrey** [comment]
Yes, this is unfortunate. We could do a hack and include an img element, just to delay the load event, but this has its own issues (e.g. it doesn't test that the css-paint-api loads the img itself).
